### DISCUSSION TITLE
"Drawing" heart-like shapes

### DISCUSSION THREAD

**Discussion by kmarler18** [tailwindlabs/tailwindcss#10197] [Help]
Hello Beautiful People!

I'm trying to implement the following heart-like shapes in my website.

![Screen Shot 2022-12-31 at 1 09 53 AM](https://user-images.githubusercontent.com/96091818/210127070-06f0f980-9c9f-4ab1-b633-00b47fda0b08.png)

What do ya'll think is the best way to implement them? I've not yet had to "draw" anything other than basic shapes using HTML and CSS so this is a first for me. 

I appreciate you (yes, YOU) ;o)
Kyle

**Comment by wongjn** [comment]
I would draw them as [SVGs](https://developer.mozilla.org/en-US/docs/Web/SVG), and display them via `background-image`.

**Reply by wongjn** [reply]
(Replying to https://github.com/tailwindlabs/tailwindcss/discussions/10197#discussioncomment-4596189)

> […] what do you think? Is this a reasonable solution?

Some of my thoughts:

- Remove the unnecessary metadata presumably generated from Sketch:
  ```diff
   <div class="h-[49rem]">
  -  <?xml version="1.0" encoding="UTF-8"?>
  -  <svg class="mx-auto mt-16 h-[187px] w-60" viewBox="0 0 240 187" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  -    <!-- Generator: Sketch 63.1 (92452) - https://sketch.com -->
  -    <title>Group 20</title>
  -    <desc>Created with Sketch.</desc>
  +  <svg class="mx-auto mt-16 h-[187px] w-60" viewBox="0 0 240 187">
       <defs>
  ```
- Remove some of the unused `id` attributes.
- Prefer `clipPath` over `mask`, since `mask` is more for translucent areas, but we have pretty much binary transparent/opaque here.
- The numerical values in the `transform` attributes could be less verbose.
- Remove the `<g>` elements by resolving the `transform` attributes into the coordinates of the child elements.
- Use HTML `<img>` elements outside of the `<svg>` instead of `<image>` elements inside. This would then allow you to use `srcset`, `sizes`, `loading=lazy`, `decoding=async` and other optimizations that `<image>` does not have (not sure what `<image>` has, since I don't use it really).
- Seems like the general heart shape could be DRY-ed into a single `<path>` and reused instead of having individual `<path>` elements for each one.

But I'd say we're deviating from the topic of Tailwind at this stage.

**Reply by kmarler18** [reply]
@wongjn 

I appreciate the feedback Justin.

**Comment by kmarler18** [comment]
Hey @wongjn !

As always, you're the man.

I want to show you what I ended up doing based on your advice, and see if you can find any glaring holes in my implementation.

The design I was given actually looks like this (not just the little hearts in the photo I included in my initial post)...

![Screen Shot 2023-01-04 at 5 18 04 PM](https://user-images.githubusercontent.com/96091818/210660754-1e124442-7aa1-426e-b386-e7d29d7d6294.png)

I'm not sure how familiar you are with the design program, Sketch, but I noticed there is a "Copy SVG Code" option.

![Screen Shot 2023-01-04 at 5 09 36 PM](https://user-images.githubusercontent.com/96091818/210660916-e2501890-8e95-4b71-abc7-e4755608e267.png)

The code that is copied when you click that button is [here](https://drive.google.com/file/d/1lwDYYX-FnhjteEJKq7b0TW5hXvCjwpxp/view?usp=sharing). The upside is that when I insert the code into my html file, it renders the image perfectly! The downside is that the code is obnoxiously long (over 1,000 pages).

After examining the code, I noticed that the majority of it appears to be a file path to an image...

![Screen Shot 2023-01-04 at 5 32 18 PM](https://user-images.githubusercontent.com/96091818/210662787-0a7e1fe5-e41c-4a3c-90f6-f402259cc772.png)

I replaced the lengthy text with a simple file path leading to the image that is downloaded on my machine, which brought the 1,000 page document down to only a few dozen lines of [code](https://play.tailwindcss.com/LHJnLUjhUA).

![Screen Shot 2023-01-04 at 5 36 33 PM](https://user-images.githubusercontent.com/96091818/210663235-070e8277-39d7-43d8-bfa7-3dbb513fdedb.png)

And it seems to work! This is it in the web browser.

![Screen Shot 2023-01-04 at 5 40 50 PM](https://user-images.githubusercontent.com/96091818/210663706-512a161a-bf48-41d3-b5bf-6b0906f48611.png)


Here's the [code in Tailwind Play](https://play.tailwindcss.com/LHJnLUjhUA) to my solution. I couldn't figure out how to include my actual image files so ignore the fact the images aren't actually the McDonald's worker. 

Anyways, long story long, what do you think? Is this a reasonable solution?

Thanks a million,
Kyle

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/10197#discussioncomment-4596354 to keep the reply chain together.
